### DISCUSSION TITLE
What happens to the unused IP addresses in a VPC?

### DISCUSSION THREAD

**Discussion by ina-stoyanova** [gruntwork-io/knowledge-base#199] [Help]
As a customer has asked, if we have a VPC with a given split of the subnets as so:

- A VPC with 10.60.0.0/18 deployed in AWS region X, with 4 AZs 
- 12 subnets are created - e.g. 10.60.0.0/23, 10.60.1.0/23, 10.60.2.0/23.....

As we can see each subnet is having approx 512 IP addresses which would mean a total of 12 subnets will have 6144 IPs. The VPC has available in total 16,384 IP addresses. 

What happens to the unused IP addresses?

**Comment by ina-stoyanova** [comment]
To my knowledge: 

A VPC with a mask of `/18` will allow up to `16,384` IP addresses (I used https://cidr.xyz/ for a calculator), and you're only using about `12subnets x 512` IP addresses from those. 
The reality is that if you haven't split your subnets in such a way to use up all the available host IP addresses those will be unallocated.

Your VPC's mask (`/18`) will still be designed to allow you to use them, but as long as you have your subnets designed in a way to consume only a small subset of those IP addresses, the ones remaining outside of an allocated VPC won't be used at all, and you can claim them later. 

Since this is all up to you, you might prefer it that way with a vision that in the future you might need to add more subnets and use up the rest of the available IP addresses. If you'd rather do that earlier than later, you might want to consider reducing the CIDR mask for your VPC, or increasing the mask for your subnets (e.g. to `/21`).

**Reply by anouarchattouna** [reply]
In addition, you need to follow AWS recommendations for [VPC and subnet sizing](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html#VPC_Sizing):

>When you create a VPC, you must specify an IPv4 CIDR block for the VPC. The allowed block size is between a `/16` netmask (65,536 IP addresses) and `/28` netmask (16 IP addresses). After you've created your VPC, you can associate secondary CIDR blocks with the VPC.

>The CIDR block of a subnet can be the same as the CIDR block for the VPC (for a single subnet in the VPC), or a subset of the CIDR block for the VPC (for multiple subnets). The allowed block size is between a `/28` netmask and `/16` netmask. If you create more than one subnet in a VPC, the CIDR blocks of the subnets cannot overlap. 

>The first four IP addresses and the last IP address in each subnet CIDR block are not available for you to use, and cannot be assigned to an instance.
